scripts/smokeTests.sh

```diff
@@ -27,60 +30,63 @@ if (( CURRENT_NODE_VERSION < REQUIRED_NODE_VERSION )); then
 fi
 
 # Autodetect project directory relative to this script's path
-PROJECT_DIR="$0"
-while [ -h "$PROJECT_DIR" ]; do
-    ls=$(ls -ld "$PROJECT_DIR")
-    link=$(expr "$ls" : '.*-> \(.*\)$')
-    if expr "$link" : '/.*' > /dev/null; then
-        PROJECT_DIR="$link"
-    else
-        PROJECT_DIR="$(dirname "$PROJECT_DIR")/$link"
-    fi
-done
-PROJECT_DIR="$(dirname "$PROJECT_DIR")/.."
-PROJECT_DIR="$(cd "$PROJECT_DIR"; pwd)"
+PROJECT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
 
-cd $PROJECT_DIR
+cd "$PROJECT_DIR"
 
 cp .env.example .env
 
 pnpm install -r
 
 pnpm build
 
+# Create temp file and ensure cleanup
 OUTFILE="$(mktemp)"
-echo $OUTFILE
+trap 'rm -f "$OUTFILE"' EXIT
+echo "Using temporary output file: $OUTFILE"
+
+# Add timeout configuration
+TIMEOUT=30
+INTERVAL=0.5
+TIMER=0
+
 (
-  # Wait for the ready message
+  # Wait for the ready message with timeout
   while true; do
```

Comment:
IMO we shouldn't be using sleep here. We should fix whatever underlying problem is causing the test to fail.

Reply:
Agreed - arbitrary sleeps can make tests both slower than necessary and introduce flakiness in my experience. I can create an issue/brainstorm a better termination signal instead